### DISCUSSION TITLE
do you follow type of id to what the underlying database suggests?

### DISCUSSION THREAD

**Discussion by kybin** [benbjohnson/wtf#21] [Q&A] [answered]
Hello, thanks for the great series of posts. I learned a lot from it.

In wtf, you use int as id type of Dial. I guess it's because sqlite creates int ids. Am I right?

MongoDB uses ObjectID which I think is only representable by string in go (unless use ObjectID directly). Would you use string id, in this case?

Thanks in advance :)

**Comment by benbjohnson** [comment, marked as answer]
Yes, I would make the ID a string on the domain object so it can stay generic. It's easy to convert between the `bson` object ID and the string representation and it makes it much simpler for the rest of your code base to just use a primitive type.

**Reply by kybin** [reply]
Understood. Thank you!
